app/main.py

```diff
@@ -1,8 +1,29 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people_list: list[dict]) -> list:
+    add_person(people_list)
+    set_spouses(people_list)
+    return list(Person.people.values())
+
+
+def add_person(people_list: list[dict]) -> None:
+    for person in people_list:
+        name = person["name"]
+        age = person["age"]
+        Person(name, age)
```

Comment:
There is no need to allocate `name` and `age`variables, pass values directly to Person. Besides that, you could make this operation within list comprehension

Reply:
fixed

Reply:
fixed